app/controllers/sessions_controller.rb

```diff
@@ -4,7 +4,12 @@ class SessionsController < ApplicationController
   def create
     user = User.update_or_create(request.env['omniauth.auth'])
     session[:id] = user.id
-    redirect_to root_path
+    flash[:success] = "Welcome #{user.full_name}!"
+    if user.address.nil?
+      redirect_to '/address/prompt'
```

Comment:
Users will be prompted with this screen every time they log in without an address.
It's not mandatory and They can skip past it.